lib/core/src/recover/model.rs

```diff
@@ -199,12 +199,15 @@ pub(crate) struct RecoveredOnchainDataChainReceive {
 impl RecoveredOnchainDataChainReceive {
     pub(crate) fn derive_partial_state(
         &self,
-        min_lockup_amount_sat: u64,
+        expected_lockup_amount_sat: Option<u64>,
         is_expired: bool,
         is_waiting_fee_acceptance: bool,
     ) -> Option<PaymentState> {
         let is_refundable = self.btc_user_lockup_address_balance_sat > 0
-            && (is_expired || self.btc_user_lockup_amount_sat < min_lockup_amount_sat); // TODO: this does not support accepting over/underpayments
+            && (is_expired
+                || expected_lockup_amount_sat.map_or(false, |expected_lockup_amount_sat| {
```

Comment:
Just confirming here. The swapper now should agree with us on over payment but worst case if not the payment will be marked as refundable and refund will only succeed after expiration.

Reply:
Yes, that's the idea. If the swapper disagrees, UX is not great because we'll show a refundable that can't be immediately refunded, but it should not happen, so I think it's fine.